plugin.video.embycon/resources/lib/views.py

```diff
@@ -0,0 +1,195 @@
+# Gnu General Public License - see LICENSE.TXT
+
+import os
+import xbmc
+import xbmcgui
+import xbmcaddon
+
+import json
+
+from simple_logging import SimpleLogging
+from translation import i18n
+
+log = SimpleLogging(__name__)
+__addon__ = xbmcaddon.Addon(id='plugin.video.embycon')
+
+
+def loadSkinDefaults():
+    defaultData = {}
+    # load current default views
+    # add a hash of xbmc.getSkinDir() to file name to make it skin specific
+    __addondir__ = xbmc.translatePath(__addon__.getAddonInfo('profile'))
+    view_list_path = os.path.join(__addondir__, "default_views.json")
+    if os.path.exists(view_list_path):
+        dataFile = open(view_list_path, 'r')
+        jsonData = dataFile.read()
+        dataFile.close()
+        defaultData = json.loads(jsonData)
+
+    return defaultData
+
+
+class DefaultViews(xbmcgui.WindowXMLDialog):
+    viewData = {}
+    sortData = {"Title": "title", "Date": "date"}
+    defaultView = {}
+    defaultSort = {}
+
+    def __init__(self, *args, **kwargs):
+        xbmcgui.WindowXMLDialog.__init__(self, *args, **kwargs)
+        log.info("WINDOW INITIALISED")
+
+    def onInit(self):
+        self.action_exitkeys_id = [10, 13]
+
+        # load skin views               
+        addonPath = __addon__.getAddonInfo('path')
+        skin_view_file = os.path.join(addonPath, "resources", "data", "skin_views.json")
+        log.info("Loading skin views form: " + skin_view_file)
+        dataFile = open(skin_view_file, 'r')
```

Comment:
This ensures proper file handling (closing in case of an error etc.)
```python
with open(skin_view_file, 'r') as dataFile:
    jsonData = dataFile.read()
defaultViewData = json.loads(jsonData)
```